.github/actions/cl2-modules/egw/manifests/egw-policy.yaml

```
@@ -16,3 +16,4 @@ spec:
     nodeSelector:
       matchLabels:
         {{StructuralData .NodeSelector}}
+  egressGateways: []
```

Why is this change necessary (and the same for similar ones)? Wouldn't requiring this modification be a breaking change of the API, that could create problems to users?

---

The problem is who this yaml files are parsed. When I leave `egressGateways` out the parsing fails because it sets that field to `null` rather than the empty array.

---

Hmm, not sure I'm understanding, why are `null` and empty array being treated differently?

---

For example, if the function [newConnDisruptCEGP()](https://github.com/cilium/cilium/blob/829016fa3bcd5d7b38ce82c5c147d628759ebf0f/cilium-cli/connectivity/check/deployment.go#L587) doesn't include it, then the [ApplyGeneric()](https://github.com/cilium/cilium/blob/829016fa3bcd5d7b38ce82c5c147d628759ebf0f/cilium-cli/connectivity/check/deployment.go#L703) method fails because the new field is set as null rather than an empty array. 

You can even try it yourself if you are curious on this behaviour. Try to remove the [ExcludedCIDRs](https://github.com/cilium/cilium/blob/829016fa3bcd5d7b38ce82c5c147d628759ebf0f/cilium-cli/connectivity/check/deployment.go#L587) field from the CEGP created by `newConnDisruptCEGP()` and you'll see how you get an error during the long running e2e tests, like `Cilium E2E Upgrade`. 

I saw a similar issue with this file.